### PR TITLE
Make a generic message event.

### PR DESCRIPTION
**Pull request by MancunianSam**
Quite often we just need to send a string of text to slack and there's
no need for the post processing that the notifications service does.

This gets rid of the secret rotation decoder and replaces it with a
generic message sender.

We can probably refactor some of the existing decoders to use this as
well but that can come later.

### CODE REVIEW

File: src/test/scala/uk/gov/nationalarchives/notifications/GenericMessageIntegrationSpec.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package uk.gov.nationalarchives.notifications
+
+import cats.implicits.catsSyntaxOptionId
+import org.scalatest.prop.TableFor7
+
+class GenericMessageIntegrationSpec extends LambdaIntegrationSpec {
+
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/notifications/GenericMessageIntegrationSpec.scala:7]: Unnecessary line break.

File: src/test/scala/uk/gov/nationalarchives/notifications/GenericMessageIntegrationSpec.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    val successfulMessageText = "A test message"
+    val twoMessages: String =
+      """ "{\"messages\":[{\"message\":\"A test message\"}, {\"message\":\"A second test message\"}]}" """
+    val failedMessageText = "A test message\\nA second test message"
+
+    Table(
+      ("description", "input", "emailBody", "slackBody", "sqsMessage", "stubContext", "slackUrl"),
+      ("a single message",
+        secretRotationEventInput(oneMessage), None, slackMsg(successfulMessageText).some, None, () => (), "/webhook"),
+      ("multiple messages",
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/notifications/GenericMessageIntegrationSpec.scala:21]: Should the description be "failed message", or something like that, as this is returning the `failedMessageText`

File: src/test/scala/uk/gov/nationalarchives/notifications/GenericMessageIntegrationSpec.scala [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        secretRotationEventInput(twoMessages), None, slackMsg(failedMessageText).some, None, () => (), "/webhook")
+    )
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/notifications/GenericMessageIntegrationSpec.scala:23]: Should this also include a "multiple message" test?

File: src/test/scala/uk/gov/nationalarchives/notifications/GenericMessageIntegrationSpec.scala [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  }
+
+  def secretRotationEventInput(rotationNotification: String): String =
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/notifications/GenericMessageIntegrationSpec.scala:26]: Should this be rename `genericEventInput` as this is now a generic message event notification? Also the param name should be changed to reflect this as well.

**Review by TomJKing — COMMENTED** [review]
Looks good. Just some minor comments. Thanks

File: src/test/scala/uk/gov/nationalarchives/notifications/GenericMessageIntegrationSpec.scala (hunk 2 above)
Comment (MancunianSam) [reply src/test/scala/uk/gov/nationalarchives/notifications/GenericMessageIntegrationSpec.scala:21]: It's the `failedMessageText` that doesn't make sense so I've changed it to `twoMessagesText`

File: src/test/scala/uk/gov/nationalarchives/notifications/GenericMessageIntegrationSpec.scala (hunk 3 above)
Comment (MancunianSam) [reply src/test/scala/uk/gov/nationalarchives/notifications/GenericMessageIntegrationSpec.scala:23]: It does already I guess if you count two as multiple. Or do you mean check for three or more messages?

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**